tests/utils/esx/esxcli.go

```diff
@@ -77,3 +78,39 @@ func WaitForExpectedState(fn getVMPowerStatus, vmName, expectedState string) boo
 	log.Printf("Timed out to poll status\n")
 	return false
 }
+
+// IsVMBackOnline checks if VM is fully back online after being restarted
+func IsVMBackOnline(vmIP, vmName string) bool {
+	isStatusOnline := WaitForExpectedState(GetVMPowerState, vmName, properties.PowerOnState)
+	if !isStatusOnline {
+		log.Printf("VM %s is not in PoweredOn state", vmName)
+		return false
+	}
+
+	isVDVSRunning := isVDVSRunning(vmIP)
+	if !isVDVSRunning {
+		log.Printf("vDVS is not running on VM [%s]. Please check if the previous VM IP [%s] has changed.", vmName, vmIP)
+		return false
+	}
+
+	return true
+}
+
+// isVDVSRunning checks if vDVS is running. This util can be useful
+// in scenarios where vm is powered-on and user wants to find out if
+// vm is fully up to be able to run docker volume commands.
+func isVDVSRunning(ip string) bool {
```

Comment:
> vDVS is also running on ESX, so it's fair enough to have esx utils to check vDVS status as well.

There should be some confusion, `isVDVSRunning` is not checking the service running on ESX instead it is checking on VM. 

On the other thing just noticed, this method is not exported that requires another refactoring when required. Where as original proposal is having in exported and the code was reviewed at #1381 and later broke it down checked through #1478 (see https://github.com/vmware/docker-volume-vsphere/pull/1478#issuecomment-310925916)